messaging/aq/src/main/java/io/helidon/messaging/connectors/aq/AqConnector.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c)  2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.messaging.connectors.aq;
+
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.literal.NamedLiteral;
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.sql.DataSource;
+
+import io.helidon.config.Config;
+import io.helidon.config.ConfigValue;
+import io.helidon.messaging.MessagingException;
+import io.helidon.messaging.connectors.jms.JmsConnector;
+import io.helidon.messaging.connectors.jms.JmsMessage;
+import io.helidon.messaging.connectors.jms.SessionMetadata;
+
+import oracle.jms.AQjmsConnectionFactory;
+import oracle.jms.AQjmsMessage;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.spi.Connector;
+
+/**
+ * MicroProfile Reactive Messaging Oracle AQ connector.
+ */
+@ApplicationScoped
+@Connector(AqConnector.CONNECTOR_NAME)
+public class AqConnector extends JmsConnector {
+
+    private static final Logger LOGGER = Logger.getLogger(AqConnector.class.getName());
+    /**
+     * Microprofile messaging Oracle AQ connector name.
+     */
+    public static final String CONNECTOR_NAME = "helidon-aq";
+
+    private static final String DATASOURCE_PROP = "data-source";
+    private static final String URL_PROP = "url";
+    private static final String USERNAME_PROP = "username";
+    private static final String PASSWORD_PROP = "password";
+
+    @Inject
+    private Instance<AQjmsConnectionFactory> connectionFactories;
+
+    /**
+     * Create new JmsConnector.
```

Comment:
Copy/paste error? JmsConnector - should be AqConnector

Reply:
thx